app/models/report.rb

```diff
@@ -3,6 +3,12 @@
 class Report < ApplicationRecord
   belongs_to :user
   has_many :comments, as: :commentable, dependent: :destroy
+  # 自身が引用されたレポートを持っていることをhas_manyで定義
+  has_many :mentioned_relation, foreign_key: 'mention_destination_report_id', class_name: 'Mention', dependent: :destroy, inverse_of: :mention_source_report
+  has_many :mentioned_reports, through: :mentioned_relation, source: :mention_source_report
+  # 自信を引用したレポートを持っていることをhas_manyで定義
```

Comment:
自信→自身ですね。あと、必要最小限の情報にするなら

```suggestion
  # 自身を引用したレポート
```

だけで十分かも？

が、そもそもコメントの必要がないかもしれませんね。
https://qiita.com/jnchito/items/f0d90af4ed44b7484103